### ISSUE TITLE
finger15.tac patch

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5250]
|<img alt="anv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| anv reported|
|-|-|
|Trac ID|trac#5250|
|Type|enhancement|
|Created|2011-09-04 13:21:48Z|

In the current version of finger15.tac 
the _read method has never been called. 
Proposal patch solves the problem.

```
Index: trunk/doc/core/howto/tutorial/listings/finger/finger15.tac
===================================================================
--- trunk/doc/core/howto/tutorial/listings/finger/finger15.tac	(revision 32505)
+++ trunk/doc/core/howto/tutorial/listings/finger/finger15.tac	(working copy)
@@ -79,6 +79,7 @@
 
 application = service.Application('finger', uid=1, gid=1)
 f = FingerService('/etc/users')
+f.setServiceParent(application)
 serviceCollection = service.IServiceCollection(application)
 internet.TCPServer(79, f.getFingerFactory()
                    ).setServiceParent(serviceCollection)
```

Attachments:

* [tutorial_finger.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/f7e/f7ebe64806b3340e6f5958d5135e367114533de1/1ed03bc8f6da6302d33deeb69c14fa1a5db75d49.patch) (5281 bytes) - added by anv on 2011-09-21 20:02:57Z - The whole patch 

<details><summary>Searchable metadata</summary>

```
trac-id__5250 5250
type__enhancement enhancement
reporter__anv anv
priority__low low
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1315142508000000 1315142508000000
changetime__1316647422000000 1316647422000000
version__None None
owner__anv anv
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="anv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_1"></a>|anv commented|
|-|-|


Conceptually this is much better:

```
Index: trunk/doc/core/howto/tutorial/listings/finger/finger15.tac
===================================================================
--- trunk/doc/core/howto/tutorial/listings/finger/finger15.tac	(revision 30844)
+++ trunk/doc/core/howto/tutorial/listings/finger/finger15.tac	(working copy)
@@ -80,6 +80,7 @@
 application = service.Application('finger', uid=1, gid=1)
 f = FingerService('/etc/users')
 serviceCollection = service.IServiceCollection(application)
+f.setServiceParent(serviceCollection)
 internet.TCPServer(79, f.getFingerFactory()
                    ).setServiceParent(serviceCollection)
 internet.TCPServer(8000, server.Site(f.getResource())
```

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_2"></a>|@thijstriemstra commented|
|-|-|

Putting the patch up for review as described in the ReviewProcess.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set owner to anv|
|-|-|

Thanks.  Due to the construction of the finger tutorial, the mistake is duplicated in every subsequent tac as well.  Would you like to attach a patch addressing all of them?

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set status to `closed`|
|-|-|

(In [32568]) Apply tutorial_finger.patch fixing the finger tutorial listings to correctly construct the service hierarchy.

Author: anv
Reviewer: exarkun
Fixes: [#5250](https://github.com/twisted/twisted/issues/5250)

Fix many finger tutorial listings, which did not add the finger service to the
service hierarchy, preventing the service from ever reading the user list.
